framework/src/modules/chain/submodules/loader.js

```diff
@@ -184,70 +185,54 @@ __private.syncTimer = function() {
 };
 
 /**
- * Gets a random peer and loads signatures from network.
- * Processes each signature from peer.
+ * Loads signatures from network.
+ * Processes each signature from the network.
  *
  * @private
  * @param {function} cb
  * @returns {setImmediateCallback} cb, err
  * @todo Add description for the params
  */
-__private.loadSignatures = function(cb) {
-	async.waterfall(
-		[
-			function(waterCb) {
-				self.getRandomPeerFromNetwork((getRandomPeerFromNetworkErr, peer) => {
-					if (getRandomPeerFromNetworkErr) {
-						return setImmediate(waterCb, getRandomPeerFromNetworkErr);
-					}
-					return setImmediate(waterCb, null, peer);
-				});
-			},
-			function(peer, waterCb) {
-				library.logger.info(`Loading signatures from: ${peer.string}`);
-				peer.rpc.getSignatures((err, res) => {
-					if (err) {
-						modules.peers.remove(peer);
-						return setImmediate(waterCb, err);
-					}
-					return library.schema.validate(
-						res,
-						definitions.WSSignaturesResponse,
-						validateErr => setImmediate(waterCb, validateErr, res.signatures)
-					);
+__private.loadSignatures = async function() {
```

Comment:
I am emphasising more on naming at some places, here also I think we can name it better `getSignaturesFromNetwork` which is more verbose and clear. I know that we used the names before, I think we can correct the naming of function and logs accordingly for more meaningful name.